dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java

```diff
@@ -150,7 +164,15 @@ public void decode() throws Exception {
 
     private void handleValue(ObjectInput in) throws IOException {
         try {
-            Type[] returnTypes = RpcUtils.getReturnTypes(invocation);
+            Type[] returnTypes = null;
+            if(useInvocationReturn) {
+                if(invocation instanceof RpcInvocation) {
+                    returnTypes = ((RpcInvocation) invocation).getReturnTypes();
+                }
+            }
```

Comment:
When calling generically, the deserialization here should be implemented by GenericFilter. If there is a generic problem, PojoUtils should be modified to solve it.